### PR TITLE
All completed besides the function on the last coding challenge

### PR DESCRIPTION
**Pull request by NikkiDunlap**
# Submitting Week 4 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [ X ] JavaScript Coding Challenges
- [ X ] Ruby Coding Challenges
- [ X ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,38 +14,166 @@
...
 
 // b) Create the function that makes the test pass.
+      // Got stuck on the function
+const shuffleWithoutFirstItem = (array) => {
+  const firstItem = array.shift();
+  const shuffled = array.sort(() => Math.random() - ???);
+  // 
+  return shuffled;
+}
+
```
Comment (chrisyhlee) [inline code-challenges.test.js:58]: So close! The thought process is great, just missing that last piece. Think about what Math.random() is giving you, as well as what is happening within the `.sort()` method. You are right at the edge here! Also, be sure to include your pseudocode, even if you got stuck at that last part! If you have further questions, don't hesitate to message me on Slack!

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,38 +14,166 @@
...
+  // Tests:       4 failed, 4 total
+  // Snapshots:   0 total
+  // Time:        0.16 s, estimated 1 s
 
 // b) Create the function that makes the test pass.
+const voteTally = (votes) => {
+  return votes.upVotes - votes.downVotes
+
+} //output:
+//   PASS  ./code-challenges.test.js
```
Comment (chrisyhlee) [inline code-challenges.test.js:122]: Super clean, love it

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,38 +14,166 @@
...
+
+    //     at Object.mergeArrays (code-challenges.test.js:113:20)
+
+    // b) Create the function that makes the test pass.
+    const mergeArrays = (...arrays) => {
+      const mergedArray = [].concat(...arrays)
+      const differentArray = Array.from(new Set(mergedArray))
+      return differentArray;
+    } //output:  
+    // PASS ./code-challenges.test.js
```
Comment (chrisyhlee) [inline code-challenges.test.js:172]: Very nice! Great use of `Array.from`, `...`, and `new Set`. Also great job on using the spread operator in the parameter field to allow as many arguments as you want!

File: interview-questions.md [hunk 4]
```diff
@@ -8,42 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 1. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: You extract data using bracket notation. As an example, if we have a hash like this. 
+veteran = { name: "Nikki", age: 30, service: "Navy" }
+We would extract the data using the below bracket notation, and it would return the key and values below.
+veteran[:name] # "Nikki"
+veteran[:age] # 30
+veteran[:service] # "Navy"
```
Comment (chrisyhlee) [inline interview-questions.md:52]: Great example here!

File: ruby-challenges.rb [hunk 5]
```diff
@@ -3,32 +3,58 @@
...
+        # Output will be the given number using string interpolation with saying "#{number} is even" or "#{number is odd}"
+
+            def even_odd(number)
+                if number.even? 
+                    "#{number} is even"
+                else number.odd? 
+                    "#{number} is odd"
+                end
+            end
+
```
Comment (chrisyhlee) [inline ruby-challenges.rb:19]: Don't need to give `else` a condition as it is a catch all for when something does not meet the condition set in the `if` statement.

File: ruby-challenges.rb [hunk 6]
```diff
@@ -3,32 +3,58 @@
...
+            # Create Ruby method named 'remove_vowels' that takes in a string as the parameter
+            # Use '.delete' method to delete all uppercase and lowercase vowels within given string
+            # Output will be the given strings with all vowels removed
+    
+                def remove_vowels(string)
+                    string.delete("aeiou" || "AEIOU")
+                end
+                p remove_vowels('Rubber Soul')
+                p remove_vowels('Sgt Pepper')
+                p remove_vowels('Abbey Road')
```
Comment (chrisyhlee) [inline ruby-challenges.rb:39]: I see your idea using the logical operator, but it's not quite working how you want it to. If you test it on `beatles_album3`, it still doesn't remove that capital A. You have the right idea, but the logical operator is actually not necessary at all!

File: ruby-challenges.rb [hunk 7]
```diff
@@ -3,32 +3,58 @@
...
+        # Output will be the given string using string interpolation with saying "#{string} is a palindrome" or "#{string} is not a palindrome"
+
+            def is_it_a_palindrome(string)
+                if string.downcase == string.downcase.reverse
+                    "#{string} is a palindrome"
+                else
+                    "#{string} is not a palindrome"
+                end
+            end
+            p is_it_a_palindrome("Racecar")
```
Comment (chrisyhlee) [inline ruby-challenges.rb:57]: Nicely done!

**Review by chrisyhlee — APPROVED** [review]
Nikki, this is great work! This assessment is displaying the Advanced Beginner level. For JavaScript 1, even if you get stuck, remember to add your pseudocode so we can better understand your thought process! I left some comments that I hope will help you with some areas, but other than that, great work! Keep going!

File: code-challenges.test.js (hunk 1 above)
Comment (NikkiDunlap) [reply code-challenges.test.js:58]: Ughhh I knew I messed this one up. Thank you for giving me insight on what to do next!

File: code-challenges.test.js (hunk 2 above)
Comment (NikkiDunlap) [reply code-challenges.test.js:122]: Thank you!

File: code-challenges.test.js (hunk 3 above)
Comment (NikkiDunlap) [reply code-challenges.test.js:172]: I was hoping that I had it right. This one stumped me for an entire day lol

File: ruby-challenges.rb (hunk 6 above)
Comment (NikkiDunlap) [reply ruby-challenges.rb:39]: I was so iffy about using that. Thank you for letting me know!

**Pull request merged**